src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt

```diff
@@ -113,22 +116,32 @@ open class IntelliJPlugin : Plugin<Project> {
     }
 
     private fun checkGradleVersion(project: Project) {
+        if (project.isBuildFeatureDisabled(BUILD_FEATURE_CHECK_GRADLE_VERSION)) {
+            return
+        }
         if (Version.parse(project.gradle.gradleVersion) < Version.parse("6.7")) {
             throw PluginInstantiationException("${IntelliJPluginConstants.NAME} requires Gradle 6.7 and higher")
         }
     }
 
     private fun checkPluginVersion(project: Project) {
+        if (project.isBuildFeatureDisabled(BUILD_FEATURE_CHECK_GRADLE_INTELLIJ_PLUGIN_VERSION)) {
```

Comment:
what about `CI`? automatically disable this check? all checks? or ability to override `CI` check default via property in `BuildFeature` enum?

Reply:
Why shouldn't that be checked on CI? If someone pushes changes to the remote without checking it locally, that could be the only way to react to an unexpected state.

Reply:
By default, we should assume CI has no network access.